ocean_lib/common/ddo/ddo.py

```diff
@@ -445,8 +445,12 @@ def is_consumable(
         if self.is_disabled:
             return ConsumableCodes.ASSET_DISABLED
 
-        if with_connectivity_check and not DataServiceProvider.check_asset_file_info(
-            self, provider_uri
+        if (
+            with_connectivity_check
+            and provider_uri
```

Comment:
Please note logic change: Connectivity check is only performed when provider_uri is set.

Reply:
The caller should pass the default provider_uri then. It shouldn't be this method's responsibility to get the default provider_uri.  Reverting the calling convention is not necessary to achieve the behavior you describe and doing so would yield an API that is harder to test and less obvious what all the inputs are.